plugins/TagFix_Brand.py

```diff
@@ -0,0 +1,106 @@
+        self.frequent_names_from_nsi = self._get_frequent_names()
+        self.brands_from_nsi = self._get_brands()
+
+    def _get_frequent_names(self):
+        nsi_url_for_names = "https://raw.githubusercontent.com/osmlab/name-suggestion-index/master/dist/names_keep.json"
+        json_str = urlread(nsi_url_for_names, 30)
+        results = json.loads(json_str)
+        return set([elem.lower() for elem in results.keys()])
+
+    def _get_brands(self):
+        nsi_url_for_brands = "https://raw.githubusercontent.com/osmlab/name-suggestion-index/master/dist/brands.json"
+        json_str = urlread(nsi_url_for_brands, 30)
+        results = json.loads(json_str)
+        additional_brands = {}
+        for brand_nsi_name, brand in results["brands"].items():
+            if "countryCodes" in brand and self.country_code.lower() not in brand["countryCodes"]:
+                continue
+            brand_nsi_name = brand_nsi_name.split("~")[0]
+            if "matchTags" in brand:
+                for additional_tag in brand["matchTags"]:
+                    nsi_key = "{}|{}".format(additional_tag, brand_nsi_name.split("|")[1])
+                    additional_brands[nsi_key.lower()] = brand
+            if "matchNames" in brand:
+                for additional_name in brand["matchNames"]:
+                    nsi_key = "{}|{}".format(brand_nsi_name.split("|")[0], additional_name)
+                    additional_brands[nsi_key.lower()] = brand
+            additional_brands[brand_nsi_name.lower()] = brand
+        return additional_brands
+
+    def node(self, data, tags):
+        if "name" in tags and not "brand" in tags:
+            for main_key in ["shop", "amenity"]:
+                if main_key in tags:
+                    nsi_key = "{}/{}|{}".format(main_key, tags[main_key], tags["name"]).lower()
+                    if nsi_key in self.frequent_names_from_nsi:
+                        if nsi_key in self.brands_from_nsi:
+                            brands_tags = self.brands_from_nsi[nsi_key]["tags"]
+                            tags_to_add = {}
+                            for tag in brands_tags:
+                                if not tags.get(tag):
+                                    tags_to_add[tag] = brands_tags[tag]
+                            return {"class": 31301, "subclass": 0,
+                                    "fix": {"+": tags_to_add}}
+                        else:
+                            return {"class": 31301, "subclass": 1,
+                                    "fix": {"+": {"brand": tags["name"]}}}
+
+    def way(self, data, tags, nds):
+        return self.node(data, tags)
+
+    def relation(self, data, tags, members):
+        return self.node(data, tags)
+
+
+###########################################################################
+
+
+class Test(TestPluginCommon):
+    def test(self):
+        a = TagFix_Brand(None)
+        a.init(None)
```

Comment:
Define a country in the test, since you use it.

```suggestion
        class _config:                                                                                             
            options = {"language": "fr"}                                                                           
        class father:                                                                                              
            config = _config()                                                                                     
        a.father = father()   
        a.init(None)
```